phpBB/phpbb/db/tools/factory.php

```diff
@@ -13,31 +13,18 @@
 
 namespace phpbb\db\tools;
 
+use Doctrine\DBAL\Connection;
+
 /**
  * A factory which serves the suitable tools instance for the given dbal
  */
 class factory
 {
 	/**
-	 * @param mixed $db_driver
-	 * @param bool $return_statements
-	 * @return \phpbb\db\tools\tools_interface
+	 * @return tools_interface
 	 */
-	public function get($db_driver, $return_statements = false)
+	public function get(Connection $connection, $return_statements = false)
```

Comment:
As discussed before, one option here would be to modify our current `dbal` to return a doctrine connection. That would keep API BC for this.

Reply:
But it means we have t ostore all the credentials in all our drivers, which is something we don't (kinda on purpose)

Reply:
or just create it on `connect()`? i don't really see why we wouldn't want to store credentials though. It doesn't seem like a security issue, as if someone has access to the memory, they can probably get the credentials anyway.

Reply:
Honestly, about API BC here I don't think we care: no one should be calling the factory manually (it is injected in the migrations).

Reply:
yes, that